File: cloud/dynamic-site/gt-cs-1331-csharp/solutions/Solution111.cs
using _4MeOnWeb.SolutionAPI;

namespace _4MeOnWeb.Solutions {
  public class Solution111 : ImmutableSolution {
    public Solution111() :
        base("1.11", "Projection") {
    }

    public override IEnumerable<string> Answers {
      get {
        long crnt_ppln = 312032486;
        int crnt_year = 2021;
        double sec_year = 365 * 24 * 60 * 60;
        double born_year = sec_year / 7;
        double death_year = sec_year / 13;
        double immgr_year = sec_year / 45;
        long dlt_year = Convert.ToInt64(Math.Round(born_year - death_year + immgr_year));

        return Enumerable.Range(1, 5)
          .Select(year =>
            $"In {crnt_year + year:d} population is {crnt_ppln + dlt_year * year:d}");
      }
    }
  }
}

File: cloud/dynamic-site/gt-cs-1331-py/app/gtcs1331/solutions/solution111.py
'''Solution 1.11 implementation'''
from gtcs1331.solution_api.solution import Solution

class Solution111(Solution):
    '''Solution 1.11'''
    id = "1.11"
    title = "Projection"

    def __answer(self):
        crnt_ppln = 312032486
        crnt_year = 2021
        sec_year = 365 * 24 * 60 * 60
        born_year = sec_year / 7
        death_year = sec_year / 13
        immgr_year = sec_year / 45
        dlt_year = round(born_year - death_year + immgr_year,0)

        return list(map(lambda year:
        f"In {int(crnt_year + year):d} population is {int(crnt_ppln + dlt_year * year):d}",
            list(range(1,6))))

    answer = property(__answer)
